odoo/tools/config.py

```diff
@@ -524,6 +533,13 @@ def _check_addons_path(self, option, opt, value, parser):
 
         setattr(parser.values, option.dest, ",".join(ad_paths))
 
+    def _check_migrations_path(self, option, opt, value, parser):
+        path = value.strip()
+        res = os.path.abspath(os.path.expanduser(path))
```

Comment:
Missing `os.path.expandvars` compared to expansion above. Do not change the expansion for addons_path to make it consistent as well, at least not for stable.